src/DataLayer/KeyMotorData/KeyMotorData.cpp

```diff
@@ -21,19 +21,21 @@ void KeyMotorData::setMotorZero(KeyMotor motorZero)
 {
     motorZero_ = motorZero;
     emit motorZeroReceived(motorZero_);
-    emit motorSetPowerReceived(((motorZero_.busVoltage() + motorOne_.busVoltage()) / 2) * ((motorZero_.busCurrent() + motorOne_.busCurrent()) / 2));
     emit motorSetCurrentReceived((motorZero_.setCurrent() + motorOne_.setCurrent()) / 2);
-    emit motorActualSpeedReceived((motorZero_.vehicleVelocity() + motorOne_.vehicleVelocity()) / 2);
     emit motorBusVoltageReceived((motorZero_.busVoltage() + motorOne_.busVoltage()) / 2);
     emit motorBusCurrentReceived((motorZero_.busCurrent() + motorOne_.busCurrent()) / 2);
+
+    emit motorZeroBusPowerReceived(motorZero_.busVoltage() * motorZero_.busCurrent());
+    emit motorActualSpeedReceived((motorZero_.vehicleVelocity() + motorOne_.vehicleVelocity()) / 2);
```

Comment:
Re-add this between `emit motorSetCurrentReceived` and `emit motorBusVoltageReceived`, so it doesn't show up in the diff